test/e2e/network_segmentation.go

```diff
@@ -365,8 +372,226 @@ var _ = Describe("Network Segmentation", func() {
+		})
+
+		Context("pod connected to UserDefinedNetwork", func() {
+			const testPodName = "test-pod-udn"
+
+			var (
+				udnInUseDeleteTimeout = 65 * time.Second
+				deleteNetworkTimeout  = 5 * time.Second
+				deleteNetworkInterval = 1 * time.Second
+			)
+
+			BeforeEach(func() {
+				By("create pod")
+				networkAttachments := []nadapi.NetworkSelectionElement{
+					{Name: testUdnName, Namespace: f.Namespace.Name},
+				}
+				cfg := podConfig(testPodName, withNetworkAttachment(networkAttachments))
+				cfg.namespace = f.Namespace.Name
+				runUDNPod(cs, f.Namespace.Name, *cfg, nil)
+			})
+
+			It("UserDefinedNetwork cannot be deleted when being used", func() {
+				By("verify UserDefinedNetwork cannot be deleted")
+				cmd := e2ekubectl.NewKubectlCommand(f.Namespace.Name, "delete", userDefinedNetworkResource, testUdnName)
+				cmd.WithTimeout(time.NewTimer(deleteNetworkTimeout).C)
+				_, err := cmd.Exec()
+				Expect(err).To(HaveOccurred(),
+					"should fail to delete UserDefinedNetwork when used")
+
+				By("verify UserDefinedNetwork associated NetworkAttachmentDefinition cannot be deleted")
+				Eventually(func() error {
+					ctx, cancel := context.WithTimeout(context.Background(), deleteNetworkTimeout)
+					defer cancel()
+					_ = nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Delete(ctx, testUdnName, metav1.DeleteOptions{})
+					_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Get(ctx, testUdnName, metav1.GetOptions{})
+					return err
+				}).ShouldNot(HaveOccurred(),
+					"should fail to delete UserDefinedNetwork associated NetworkAttachmentDefinition when used")
+
+				By("delete test pod")
+				err = cs.CoreV1().Pods(f.Namespace.Name).Delete(context.Background(), testPodName, metav1.DeleteOptions{})
+				Expect(err).ToNot(HaveOccurred())
+
+				By("verify UserDefinedNetwork has been deleted")
+				Eventually(func() error {
+					_, err := e2ekubectl.RunKubectl(f.Namespace.Name, "get", userDefinedNetworkResource, testUdnName)
+					return err
+				}, udnInUseDeleteTimeout, deleteNetworkInterval).Should(HaveOccurred(),
+					"UserDefinedNetwork should be deleted following test pod deletion")
+
+				By("verify UserDefinedNetwork associated NetworkAttachmentDefinition has been deleted")
+				Eventually(func() bool {
+					_, err := nadClient.NetworkAttachmentDefinitions(f.Namespace.Name).Get(context.Background(), testUdnName, metav1.GetOptions{})
+					return err != nil && kerrors.IsNotFound(err)
+				}, deleteNetworkTimeout, deleteNetworkInterval).Should(BeTrue(),
+					"NetworkAttachmentDefinition should be deleted following UserDefinedNetwork deletion")
+			})
+		})
+
+		Context("primary network exist", func() {
+			const (
+				primaryNadName = "cluster-primary-net"
+				primaryUdnName = "primary-net"
+			)
+
+			BeforeEach(func() {
+				By("create primary network NetworkAttachmentDefinition")
+				primaryNetNad := newPrimaryNetAttachDef(f.Namespace.Name, primaryNadName)
```

Comment:
I think you can reuse these functions similar to existing tests
```
netConfig := newNetworkAttachmentConfig(c)
nad := generateNAD(netConfig)
```

Reply:
Done